### ISSUE TITLE
Disconnect from Sync screen does not receive keyboard focus when opened

### ISSUE THREAD

**Issue by shane-tomlinson** [mozilla/fxa-content-server#6157]
STR

1. Create an account, attach to Sync.
2. Open /settings
3. Open Devices & apps
4. Click on the "Disconnect" button next to a device.
5. Try to tab around using the keyboard. The page underneath scrolls.

This makes it difficult for screen reader users to use the functionality.

![disconnect-device-focus](https://user-images.githubusercontent.com/848085/39572182-2173fe5e-4ec6-11e8-9088-7f11d246426b.gif)

**Comment by farhan787** [comment]
@shane-tomlinson I want to fix this.

**Comment by DraciVik** [comment]
Hi @shane-tomlinson I am an Outreachy applicant I want to give this a try. Can I give it a go?

**Comment by shane-tomlinson** [comment]
@farhan787 go for it!

@DraciVik This one is already spoken for.

**Comment by farhan787** [comment]
@shane-tomlinson thanks

**Comment by farhan787** [comment]
@shane-tomlinson could you please tell me what do you mean by attach to sync in the first step to reproduce this ?

**Comment by karansapolia** [comment]
I think this is a common issue with all modals. Tried it with all the 'modal-panel' classes and noticed no one received tab focus after opening. I think this is a more general issue as this is the same problem as in #6594. I think even the solution for these problems would be a change in one common file. @shane-tomlinson  @lmorchard @ryanfeeley Thoughts and inputs requested.

**Comment by farhan787** [comment]
@karansapolia you are right, this is a common issue with all modals.

**Comment by shane-tomlinson** [comment]
@karansapolia you are correct that this could be handled universally. The [settings-panel-mixin](https://github.com/mozilla/fxa-content-server/blob/4cfac43f4df7d7006fb7df699b3750375dbf08e9/app/scripts/views/mixins/settings-panel-mixin.js#L25) already has some logic to autofocus [elements with the autofocus attribute from openPanel](https://github.com/mozilla/fxa-content-server/blob/4cfac43f4df7d7006fb7df699b3750375dbf08e9/app/scripts/views/mixins/settings-panel-mixin.js#L45), maybe that can be extracted into a common area and included in both settings-panel-mixin and [modal-settings-panel-mixin](https://github.com/mozilla/fxa-content-server/blob/4cfac43f4df7d7006fb7df699b3750375dbf08e9/app/scripts/views/mixins/modal-settings-panel-mixin.js#L1) using a [dependsOn declaration](https://github.com/mozilla/fxa-content-server/blob/4cfac43f4df7d7006fb7df699b3750375dbf08e9/app/scripts/lib/user-agent-mixin.js#L18). Once that's added, `autofocus` attributes would need to be added to the appropriate elements within the templates.

**Comment by karansapolia** [comment]
That is helpful @shane-tomlinson, thank you, but I am focusing more on building a focus trap in modal-panel-mixin, because we will not like keyboard focus to escape the modal until user presses escape. It would be better if the keyboard focus will cycle through the modal elements than to escape it and move to background elements. That would become confusing for user. Your thoughts?
